internal/providers/openstack/openstack.go

```diff
@@ -91,9 +95,31 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 		return fetchConfigFromMetadataService(f)
 	})
 
-	<-ctx.Done()
-	if ctx.Err() == context.DeadlineExceeded {
-		f.Logger.Info("neither config drive nor metadata service were available in time. Continuing without a config...")
+	errCount := 0
+
+	// function literal to check both ctx.Done() as well as the amount
+	// of errors in the errChan to allow for non-blocking checks
+	waitUntilSuccessOrAllFailures := func() bool {
```

Comment:
Function is misnamed; it doesn't `waitUntil`, but the caller has to call it in a loop.  Actually, does this need to be in a helper function at all?

Reply:
Re-worked to not use a literal